modules/video/src/tracking/tracker_nano.cpp

```diff
@@ -0,0 +1,361 @@
+    Mat predY2 = trackState.grid2searchY + bboxPred.row(3).reshape(0, {trackState.scoreSize, trackState.scoreSize});
+
+    // size penalty
+    // scale penalty
+    Mat sc = sizeCal(predX2 - predX1, predY2 - predY1)/sizeCal(trackState.targetPos[0], trackState.targetPos[1]);
+    elementMax(sc);
+
+    // ratio penalty
+    float ratioVal = trackState.targetSz[0] / trackState.targetSz[1];
+
+    Mat ratioM(trackState.scoreSize, trackState.scoreSize, CV_32FC1, Scalar::all(ratioVal));
+    Mat rc = ratioM / ((predX2 - predX1) / (predY2 - predY1));
+    elementMax(rc);
+
+    Mat penalty;
+    exp(((rc.mul(sc) - 1) * trackState.penaltyK * (-1)), penalty);
+    Mat pscore = penalty.mul(score);
+
+    // Window penalty
+    pscore = pscore * (1.0 - trackState.windowInfluence) + trackState.hanningWindow * trackState.windowInfluence;
+
+    // get Max
+    int bestID[2] = { 0, 0 };
+    minMaxIdx(pscore, 0, 0, 0, bestID);
+
+    trackState.tracking_score = pscore.at<float>(bestID);
+
+    float x1Val = predX1.at<float>(bestID);
+    float x2Val = predX2.at<float>(bestID);
+    float y1Val = predY1.at<float>(bestID);
+    float y2Val = predY2.at<float>(bestID);
+
+    float predXs = (x1Val + x2Val)/2;
+    float predYs = (y1Val + y2Val)/2;
+    float predW = (x2Val - x1Val)/scale_z;
+    float predH = (y2Val - y1Val)/scale_z;
+
+    float diffXs = (predXs - trackState.instanceSize / 2) / scale_z;
+    float diffYs = (predYs - trackState.instanceSize / 2) / scale_z;
+
+    trackState.targetSz[0] /= scale_z;
+    trackState.targetSz[1] /= scale_z;
+
+    float lr = penalty.at<float>(bestID) * score.at<float>(bestID) * trackState.lr;
+
+    float resX = trackState.targetPos[0] + diffXs;
+    float resY = trackState.targetPos[1] + diffYs;
+    float resW = predW * lr + (1 - lr) * trackState.targetSz[0];
+    float resH = predH * lr + (1 - lr) * trackState.targetSz[1];
+
+    resX = std::max(0.f, std::min((float)trackState.imgSize.width, resX));
+    resY = std::max(0.f, std::min((float)trackState.imgSize.height, resY));
+    resW = std::max(10.f, std::min((float)trackState.imgSize.width, resW));
+    resH = std::max(10.f, std::min((float)trackState.imgSize.height, resH));
+
+    trackState.targetPos[0] = resX;
+    trackState.targetPos[1] = resY;
+    trackState.targetSz[0] = resW;
+    trackState.targetSz[1] = resH;
+
+    // convert center to Rect2f.
+    Rect2f resBox = { resX - resW/2, resY - resH/2, resW, resH};
+    boundingBoxRes = resBox;
+    return true;
+}
+
+float TrackerNanoImpl::getTrackingScore()
+{
+    return trackState.tracking_score;
+}
+
+Ptr<TrackerNano> TrackerNano::create(const TrackerNano::Params& parameters)
+{
+    return makePtr<TrackerNanoImpl>(parameters);
+}
+
+#else  // OPENCV_HAVE_DNN
+Ptr<TrackerNano> TrackerNano::create(const TrackerNano::Params& parameters)
+{
+    (void)(parameters);
+    CV_Error(cv::Error::StsNotImplemented, "to use GOTURN, the tracking module needs to be built with opencv_dnn !");
```

Comment:
to use GOTURN -> to use Nano tracker
the tracking module -> the video module

Reply:
Thanks for code reviewing, fixed.

Reply:
Thanks for code reviewing, fixed.